src/actions/VaccineActions.js

```diff
@@ -0,0 +1,74 @@
+/**
+ * mSupply Mobile
+ * Sustainable Solutions (NZ) Ltd. 2021
+ */
+
+import { ToastAndroid } from 'react-native';
+import { PermissionSelectors } from '../selectors/permission';
+import selectScannedSensors from '../selectors/vaccine';
+import { PermissionActions } from './PermissionActions';
+import BleService from '../bluetooth/BleService';
+import { syncStrings } from '../localization/index';
+import { UIDatabase } from '../database/index';
+
+export const VACCINE_ACTIONS = {
+  SCAN_START: 'Vaccine/sensorScanStart',
+  SCAN_STOP: 'Vaccine/sensorScanStop',
+  SENSOR_FOUND: 'Vaccine/sensorFound',
+};
+
+const scanStart = () => ({ type: VACCINE_ACTIONS.SCAN_START });
+const scanStop = () => ({ type: VACCINE_ACTIONS.SCAN_STOP });
+const sensorFound = macAddress => ({ type: VACCINE_ACTIONS.SENSOR_FOUND, payload: { macAddress } });
+
+const startSensorScan = () => async (dispatch, getState) => {
+  const bluetoothEnabled = PermissionSelectors.bluetooth(getState());
+  const locationPermission = PermissionSelectors.location(getState());
+  // Ensure the correct permissions before initiating a new sync process.
+  if (!bluetoothEnabled) await dispatch(PermissionActions.requestBluetooth());
+  if (!locationPermission) await dispatch(PermissionActions.requestLocation());
+
+  if (!bluetoothEnabled) {
+    ToastAndroid.show(syncStrings.please_enable_bluetooth, ToastAndroid.LONG);
+    return null;
+  }
+
+  if (!locationPermission) {
+    ToastAndroid.show(syncStrings.grant_location_permission, ToastAndroid.LONG);
+    return null;
+  }
+
+  dispatch(scanForSensors());
+
+  // Scan will continue running until it is stopped...
+  return null;
+};
+
+const scanForSensors = () => async (dispatch, getState) => {
+  dispatch(scanStart());
+
+  const deviceCallback = device => {
+    const { id } = device;
+
+    if (id) {
+      const alreadyFound = selectScannedSensors(getState());
+      const alreadySaved = UIDatabase.get('Sensor', id, 'macAddress');
+
+      if (!alreadyFound.includes(id) && !alreadySaved) {
```

Comment:
aha! you've cleverly pre-empted a comment I made in the reducer!
Or perhaps I've reviewed in the wrong order. anyhoo.. the reducer is taking the `id` here and calling it `macAddress` - would you mind changing the param in reducer to be `id`?
or save the whole `device`? that would get around my other naming concern, about storing an address/id in a collection called `sensors`

Reply:
`get` can also return `null` https://github.com/sussol/react-native-database/blob/81e18edcfe935d796412fa8ffcae6ba69c2a4e94/src/Database.js#L84-L87

and `null.includes()` throws - suggest `?.includes()`

Reply:
Cool thanks for the repo link, that helps clarify things! In this instance I think `alreadyFound` is from the state which is empty array by default but agree better to be safe 👍. I also renamed to `alreadyScanned` because this made me reconsider the original naming hehe.  6b603bf7